See the error on the version update job

```
2024-09-27 17:52:43,428 ERROR    webservices_dispatch_action.version_updater || error while trying to get PR title for conda-forge/cf-autotick-bot-test-package-feedstock#483
Traceback (most recent call last):
  File "/home/runner/work/_actions/conda-forge/webservices-dispatch-action/version-update-pr-title/webservices_dispatch_action/version_updater.py", line 129, in update_pr_title
    gh = github.Github(os.environ["GH_TOKEN"])
                       ~~~~~~~~~~^^^^^^^^^^^^
  File "<frozen os>", line 679, in __getitem__
KeyError: 'GH_TOKEN'
```

https://github.com/conda-forge/cf-autotick-bot-test-package-feedstock/actions/runs/11075355037/job/30776117735

This will go away once we remove the subprocess call.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -12,7 +12,6 @@ packages = ["webservices_dispatch_action"]
 
 [project.scripts]
 run-webservices-dispatch-action = "webservices_dispatch_action.__main__:main"
-run-webservices-dispatch-action-version-updater = "webservices_dispatch_action.version_updater:main"
 
 [tool.ruff.lint]
 select = ["E", "F", "I", "W"]

--- a/webservices_dispatch_action/__main__.py
+++ b/webservices_dispatch_action/__main__.py
@@ -24,6 +24,7 @@ from webservices_dispatch_action.rerendering import (
     rerender,
 )
 from webservices_dispatch_action.utils import comment_and_push_if_changed
+from webservices_dispatch_action.version_updater import update_pr_title, update_version
 
 LOGGER = logging.getLogger(__name__)
 
@@ -158,32 +159,14 @@ def main():
 
                 # update version
                 _pull_docker_image()
-                curr_head = git_repo.active_branch.commit
-                cmd = [
-                    "run-webservices-dispatch-action-version-updater",
-                    "--feedstock-dir",
-                    feedstock_dir,
-                    "--repo-name",
+                LOGGER.info(
+                    "Running version update for %s with input_version %s",
                     repo_name,
-                    "--pr-number",
-                    pr_num,
-                ]
-                if input_version:
-                    cmd += ["--input-version", input_version]
-                LOGGER.info(f"Running command {' '.join(cmd)}")
-                ret = subprocess.run(
-                    cmd,
-                    env=os.environ,
+                    input_version,
                 )
-                if ret.returncode != 0:
-                    version_error = True
-                    version_changed = False
-                elif git_repo.active_branch.commit == curr_head:
-                    version_error = False
-                    version_changed = False
-                else:
-                    version_error = False
-                    version_changed = True
+                version_changed, version_error, found_version = update_version(
+                    feedstock_dir, repo_name, input_version=input_version
+                )
 
                 version_push_error = comment_and_push_if_changed(
                     action="update the version",
@@ -242,6 +225,15 @@ def main():
                                 rerender_error,
                             ),
                         )
+                    if found_version:
+                        LOGGER.info(
+                            "Updating PR title for %s#%s with version=%s",
+                            repo_name,
+                            pr_num,
+                            found_version,
+                        )
+                        update_pr_title(repo_name, pr_num, found_version)
+
         elif event_data["action"] == "lint":
             pr_num = int(event_data["client_payload"]["pr"])
             repo_name = event_data["repository"]["full_name"]

--- a/webservices_dispatch_action/version_updater.py
+++ b/webservices_dispatch_action/version_updater.py
@@ -2,9 +2,7 @@ import logging
 import os
 import pprint
 import subprocess
-import sys
 
-import click
 import conda_forge_tick.update_recipe
 import github
 from conda.models.version import VersionOrder
@@ -15,7 +13,6 @@ from conda_forge_tick.update_upstream_versions import (
     get_latest_version,
 )
 from conda_forge_tick.utils import setup_logging
-from git import Repo
 
 setup_logging()
 
@@ -156,57 +153,3 @@ def update_pr_title(
             )
             return False, True
     return False, False
-
-
-@click.command()
-@click.option(
-    "--feedstock-dir",
-    required=True,
-    type=str,
-    help="The directory of the feedstock",
-)
-@click.option(
-    "--repo-name",
-    required=True,
-    type=str,
-    help="The name of the repository (as '<owner>/<name>')",
-)
-@click.option(
-    "--input-version",
-    required=False,
-    type=str,
-    default=None,
-    help="The version to update to",
-)
-@click.option(
-    "--pr-number",
-    required=False,
-    type=int,
-    default=None,
-    help="PR number. Needed for PR title updates.",
-)
-def main(
-    feedstock_dir,
-    repo_name,
-    input_version=None,
-    pr_number=None,
-):
-    git_repo = Repo(feedstock_dir)
-
-    _, version_error, found_version = update_version(
-        git_repo,
-        repo_name,
-        input_version=input_version,
-    )
-
-    if version_error:
-        sys.exit(1)
-
-    if input_version is None and pr_number is not None and found_version:
-        # We had to find the latest versions ourselves, so now we should
-        # update the generic PR title to refelect the found version
-        _, title_error = update_pr_title(repo_name, pr_number, found_version)
-        if title_error:
-            sys.exit(1)
-
-    sys.exit(0)

--- a/webservices_dispatch_action/version_updater.py
+++ b/webservices_dispatch_action/version_updater.py
@@ -205,9 +205,7 @@ def main(
     if input_version is None and pr_number is not None and found_version:
         # We had to find the latest versions ourselves, so now we should
         # update the generic PR title to refelect the found version
-        _, title_error = update_pr_title(
-            repo_name, pr_number, found_version
-        )
+        _, title_error = update_pr_title(repo_name, pr_number, found_version)
         if title_error:
             sys.exit(1)
 

--- a/webservices_dispatch_action/version_updater.py
+++ b/webservices_dispatch_action/version_updater.py
@@ -25,6 +25,9 @@ LOGGER = logging.getLogger(__name__)
 def update_version(
     git_repo, repo_name, input_version=None
 ) -> tuple[bool, bool, str | None]:
+    """
+    Returns [whether version changed, errors occurred, new version found]
+    """
     name = os.path.basename(repo_name).rsplit("-", 1)[0]
     LOGGER.info("using feedstock name %s for repo %s", name, repo_name)
 
@@ -124,7 +127,12 @@ def update_version(
     return True, False, new_version
 
 
-def update_pr_title(repo_name: str, pr_number: int, found_version: str) -> bool:
+def update_pr_title(
+    repo_name: str, pr_number: int, found_version: str
+) -> tuple[bool, bool]:
+    """
+    Returns [whether title changed, errored]
+    """
     try:
         gh = github.Github(os.environ["GH_TOKEN"])
         repo = gh.get_repo(repo_name)
@@ -135,18 +143,19 @@ def update_pr_title(repo_name: str, pr_number: int, found_version: str) -> bool:
             repo_name,
             pr_number,
         )
-        return False
+        return False, True
     if pr.title == "ENH: update package version":  # user didn't change the default
         try:
             pr.edit(title=f"{pr.title} to {found_version}")
+            return True, False
         except Exception:
             LOGGER.exception(
                 "error while trying to change PR title for %s#%s",
                 repo_name,
                 pr_number,
             )
-            return False
-    return True
+            return False, True
+    return False, False
 
 
 @click.command()
@@ -193,11 +202,13 @@ def main(
     if version_error:
         sys.exit(1)
 
-    if input_version is None and found_version and pr_number is not None:
+    if input_version is None and pr_number is not None and found_version:
         # We had to find the latest versions ourselves, so now we should
         # update the generic PR title to refelect the found version
-        ok = update_pr_title(repo_name, pr_number, found_version)
-        if not ok:
+        _, title_error = update_pr_title(
+            repo_name, pr_number, found_version
+        )
+        if title_error:
             sys.exit(1)
 
     sys.exit(0)

